src/algebra/order/rearrangement.lean

```diff
@@ -212,3 +222,236 @@ lemma antivary.sum_mul_le_sum_comp_perm_mul (hfg : antivary f g) :
+        { exact h.1 } } },
+    { rcases h with ⟨k, hk, b, hbs, h⟩,
+      rw h,
+      refine ⟨by simpa using hbs, _⟩,
+      rw equiv.perm.subtype_congr.left_apply,
+      simp only [coe_mem],
+      exact hbs } }
+end
+
+-- where to move?
+lemma sum_mul_sum_eq_sum_perm [decidable_eq ι] (s : finset ι) (π : perm s) (hπ : π.is_cycle)
+  (hπs : π.support = univ) : (∑ i in s, f i) • (∑ i in s, g i) =
+    ∑ k in finset.range s.card, (∑ i in s, f i • g ((π^k).subtype_congr (equiv.refl _) i)) :=
+begin
+  rw [finset.sum_smul_sum, finset.product_self_eq_range_bUnion_perm π hπ hπs, finset.sum_bUnion _],
+  { apply finset.sum_congr rfl,
+    intros x hx,
+    rw finset.sum_bUnion _,
+    { apply finset.sum_congr rfl,
+      exact λ y hy, sum_singleton },
+    { rintros a ha b hb hab ⟨c, d⟩ h,
+      simp only [inf_eq_inter, mem_inter, mem_singleton, prod.mk.inj_iff] at h,
+      exfalso, apply hab,
+      rw [← h.1.1, h.2.1] } },
+    { rintros a ha b hb hab ⟨c, d⟩ h,
+      simp only [inf_eq_inter, mem_inter, mem_bUnion, mem_singleton, prod.mk.inj_iff,
+        exists_prop] at h,
+      rcases h with ⟨⟨e, he⟩, ⟨f, hf⟩⟩,
+      have h : ((π ^ a).subtype_congr (equiv.refl {a // a ∉ s})) c =
+        ((π ^ b).subtype_congr (equiv.refl {a // a ∉ s})) c,
+      { conv_lhs { rw [he.2.1, ← he.2.2, hf.2.2, ← hf.2.1] } },
+      have hc : c ∈ s := by simp only [he.2.1, he.1],
+      replace h : (π ^ a) ⟨c, hc⟩ = (π ^ b) ⟨c, hc⟩,
+      { rw [equiv.perm.subtype_congr.left_apply _] at h,
+        swap, exact hc,
+        rw [equiv.perm.subtype_congr.left_apply _] at h,
+        swap, exact hc,
+        simp only [← @subtype.coe_inj ι _ _, h] },
+      replace h : π ^ a = π ^ b,
+      { rw is_cycle.pow_eq_pow_iff hπ,
+        exact ⟨⟨c, hc⟩, by simp only [hπs, mem_univ], h⟩ },
+      simp only [pow_eq_pow_iff_modeq, equiv.perm.order_of_is_cycle hπ, hπs,
+        univ_eq_attach, card_attach] at h,
+      exfalso,
+      apply hab,
+      rw [nat.modeq.eq_of_modeq_of_abs_lt h _],
+      rw abs_sub_lt_iff,
+      simp only [mem_coe, mem_range] at *,
+      split; linarith },
+end
+
+end to_move
+
+/-! ### Chebyshev inequality -/
+
+/-! ### Scalar multiplication versions -/
+
+section smul
+variables [linear_ordered_ring α] [linear_ordered_add_comm_group β] [module α β]
+  [ordered_smul α β] {s : finset ι} {σ : perm ι} {f : ι → α} {g : ι → β}
+
+
+
+/-- **Chebyshev Inequality**: When `f` and `g` vary together, the scalar product of their sum is
+less than the size of the set times their scalar product. -/
+lemma monovary_on.sum_smul_sum_le_card_smul_sum (hfg : monovary_on f g s) :
+  (∑ i in s, f i) • (∑ i in s, g i) ≤ s.card • (∑ i in s, f i • g i) :=
+begin
+  obtain hs | hs := lt_or_ge s.card 2,
+  { interval_cases s.card,
+    { simp only [finset.card_eq_zero.mp h, finset.sum_empty, smul_zero] },
+    { cases (card_eq_one.mp h) with a ha,
+      simp only [ha, sum_singleton, card_singleton, one_smul] }},
+  { classical,
+    set π : perm s :=
```

Comment:
Instead of defining `π : perm s` and then later using `(π ^ k).subtype_congr (equiv.refl _)`, would it be easier to just define `π : perm ι`, and then use `π ^ k`?

Reply:
I see why this would take some extra work: there's no API for finitely-supported permutations outside of permutations on `fintype`s. I suspect that a few lemmas about cycles only supported on a `finset` may still be an easier way to do this.